fenix/app/src/main/java/org/mozilla/fenix/onboarding/JunoOnboardingFragment.kt

```diff
@@ -111,66 +133,113 @@ class JunoOnboardingFragment : Fragment() {
-                    pageType = JunoOnboardingPageType.NOTIFICATION_PERMISSION,
-                )
+                notificationPage?.sequencePosition()?.let {
+                    requireComponents.notificationsDelegate.requestNotificationPermission()
+                    telemetryRecorder.onNotificationPermissionClick(
+                        sequenceId = pagesTelemetrySequenceId,
+                        sequencePosition = it,
+                    )
+                }
             },
             onSkipNotificationClick = {
-                telemetryRecorder.onSkipTurnOnNotificationsClick(
-                    onboardingPageTypeList.telemetrySequenceId(),
-                    JunoOnboardingPageType.NOTIFICATION_PERMISSION,
-                )
+                notificationPage?.sequencePosition()?.let {
+                    telemetryRecorder.onSkipTurnOnNotificationsClick(
+                        sequenceId = pagesTelemetrySequenceId,
+                        sequencePosition = it,
+                    )
+                }
             },
-            onFinish = { pageType ->
-                onFinish(onboardingPageTypeList.telemetrySequenceId(), pageType)
+            onFinish = {
+                onFinish(
+                    sequenceId = pagesTelemetrySequenceId,
+                    sequencePosition = it.sequencePosition(),
+                )
             },
-            onImpression = { pageType ->
+            onImpression = {
                 telemetryRecorder.onImpression(
-                    onboardingPageTypeList.telemetrySequenceId(),
-                    pageType,
+                    sequenceId = pagesTelemetrySequenceId,
+                    pageType = it.type,
+                    sequencePosition = it.sequencePosition(),
                 )
             },
         )
     }
 
-    private fun onFinish(sequenceId: String, pageType: JunoOnboardingPageType) {
+    private fun onFinish(sequenceId: String, sequencePosition: Int) {
         requireComponents.fenixOnboarding.finish()
         findNavController().nav(
             id = R.id.junoOnboardingFragment,
             directions = JunoOnboardingFragmentDirections.actionOnboardingHome(),
         )
         telemetryRecorder.onOnboardingComplete(
             sequenceId = sequenceId,
-            pageType = pageType,
+            sequencePosition = sequencePosition,
         )
     }
 
-    private fun onboardingPageTypeList(context: Context): List<JunoOnboardingPageType> =
-        buildList {
-            add(JunoOnboardingPageType.DEFAULT_BROWSER)
-            add(JunoOnboardingPageType.SYNC_SIGN_IN)
-            if (shouldShowNotificationPage(context)) {
-                add(JunoOnboardingPageType.NOTIFICATION_PERMISSION)
+    private fun pagesToDisplay(context: Context): List<OnboardingPageUiData> {
+        val allCards: Map<String, OnboardingCardData> =
+            FxNimbus.features.junoOnboarding.value().cards
+        return allCards.values
+            .filter {
```

Comment:
Could remove the variable declaration and use, `FxNimbus.features.junoOnboarding.value().cards.values`, also making the function body an expression. 

I also feel we could extract this out of the fragment and add tests for this.